### PR TITLE
Ported example kernel_linear.py

### PR DESCRIPTION
**Pull request by jonpsy**
I'm having some doubts please help me clarify that. In kernel_linear.py, it has ```kernel.set_normalizer(AvgDiagKernelNormalizer(1.2))``` so I created ```Normalizer avg_diag_normalizer = normalizer("AvgDiagKernelNormalize") ``` which I think may be wrong. Could you guide me from here? :)

### CODE REVIEW

**Comment by jonpsy** [comment]
@vigsterkr , @gf712  Mind reviewing my porting example attempt ? :)

File: examples/meta/src/kernel/linear.sg [hunk 1]
```diff
@@ -0,0 +1,21 @@
+File traindat = '../data/fm_train_real.dat'
```
Comment (vigsterkr) [inline examples/meta/src/kernel/linear.sg]: in all the available examples these lines are different. see for example:
https://github.com/shogun-toolbox/shogun/blob/develop/examples/meta/src/binary/domainadaptationsvm.sg

but any examples currently in the repo...

**Review by vigsterkr — CHANGES_REQUESTED** [review]
plz make sure that things work locally before you actually send in a PR...

File: examples/meta/src/kernel/linear.sg (hunk 1 above)
Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg]: I'm sorry, I fixed that. Now I'm stuck at this part 

```Normalizer avg_diag_normalizer = normalizer("AvgDiagKernelNormalize")``` 

make meta_example throws this error 

```translate.TranslationFailure: 'Failed to obtain include path for Cnormalizer or normalizer or Cnormalizer or normalizer' ```.

Comment (vigsterkr) [reply examples/meta/src/kernel/linear.sg]: yes unfortunately kernel normalizers are not yet part of the base interface. there's a PR for this https://github.com/shogun-toolbox/shogun/pull/4671 that should be rebased and merged. until then you can just create the class itself: 
```
AvgDiagKernelNormalize kernel_normalizer = AvgDiagKernelNormalize()
```

Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg]: It still throws the same error. I presume that once that PR is merged, make meta_examples shouldn't throw that error, right?

Comment (vigsterkr) [reply examples/meta/src/kernel/linear.sg]: i'm not so sure what you mean... the very same error it should not throw... that pr needs to be fixed.

File: doc/readme/EXAMPLES.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ For details, see `CMakeLists.txt` in `examples/meta/` for details, `generate.py`
 It is extremely simple to add a new example: simple create another `*.sg` file.
 We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the new system -- a copy paste [entrance task](https://github.com/shogun-toolbox/shogun/issues/3555).
 
-If you porting an example is great, even better is when it comes with integration testing data of the numerical output, as described in [DEVELOPING.md](DEVELOPING.md#testing).
+If you are porting a new example, great! Better yet, add the [data](https://github.com/shogun-toolbox/shogun-data) so that we can do integration tests automatically as described in [DEVELOPING.md](DEVELOPING.md#testing).
```
Comment (vigsterkr) [inline doc/readme/EXAMPLES.md]: this should not be the part  of this pr....

**Review by vigsterkr — CHANGES_REQUESTED** [review]

File: examples/meta/src/kernel/linear.sg (hunk 1 above)
Comment (vigsterkr) [reply examples/meta/src/kernel/linear.sg]: not you should just simply update this PR with your latest changes to be able to discuss further... plz remove the parts i've mentioned above

File: examples/meta/src/kernel/linear.sg [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+File testdat = '../data/fm_test_real.dat'
+
+#![create_features]
+Features feats_train=features(train_fname)
+Features feats_test=features(test_fname)
+#![create_features]
+
+#![create_kernel]
+Kernel linear_kernel=kernel("LinearKernel")
+Normalizer avg_diag_normalizer = normalizer("AvgDiagKernelNormalize")
```
Comment (vigsterkr) [inline examples/meta/src/kernel/linear.sg]: KernelNormalizer avg_diag_normalizer = AvgDiagKernelNormalizer()

File: examples/meta/src/kernel/linear.sg (hunk 1 above)
Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg]: SImilar error I meant, 

```translate.TranslationFailure: 'Failed to obtain include path for CAvgDiagKernelNormalize or AvgDiagKernelNormalize or CAvgDiagKernelNormalize or AvgDiagKernelNormalize'```

Comment (vigsterkr) [reply examples/meta/src/kernel/linear.sg]: plz check the class name....

File: examples/meta/src/kernel/linear.sg (hunk 3 above)
Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg]:
``` 
   AvgDiagKernelNormalize kernel_normalizer = AvgDiagKernelNormalize() 
    linear_kernel.set_normalizer(kernel_normalizer(1.2))
```

Comment (vigsterkr) [reply examples/meta/src/kernel/linear.sg]: check spelling plz....

Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg]:
```
KernelNormalizer avg_diag_normalizer = AvgDiagKernelNormalizer()
linear_kernel.set_normalizer(avg_diag_normalizer(1.2))
```
Correct?

> KernelNormalizer avg_diag_normalizer = AvgDiagKernelNormalizer()

Comment (vigsterkr) [reply examples/meta/src/kernel/linear.sg]: @NanuSai try to compile and run it locally....?

Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg]: made the change, ran make meta_examples  
got this
```
translate.TranslationFailure: 'Failed to obtain include path for Cavg_diag_normalizer or avg_diag_normalizer or Cavg_diag_normalizer or avg_diag_normalizer'
```

Comment (NanuSai) [reply examples/meta/src/kernel/linear.sg]: I did a complete overhaul, pulled the latest develop branch, created a new branch as it's offshoot. I would like to create a new PR and close down this one, I hope you won't mind. Really sorry for the inconvenience caused. 

I've also noticed there has been some changes in .sg file porting and is now referred as ".sg.in". I'll make sure my new PR accounts for it.

**Comment by jonpsy** [comment]
I've made very basic errors and didn't followed PR guidelines properly which have caused major inconvenience. I'm closing this PR and making a new one that would adhere to the standards requested.

**Comment by vigsterkr** [comment]
you dontr need to close a pr... just reset your branch and that's all. plz read about git and github.

File: examples/meta/src/kernel/linear.sg.in [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+Features feats_train=features(traindat)
+Features feats_test=features(testdat)
+#![create_features]
+
+#![create_kernel]
+Kernel linear_kernel=kernel("LinearKernel")
+KernelNormalizer avg_diag_normalizer = AvgDiagKernelNormalizer()
+linear_kernel.set_normalizer(avg_diag_normalizer(1.2))
+linear_kernel.init(feats_train, feats_train)
+#![create_kernel]
```
Comment (jonpsy) [inline examples/meta/src/kernel/linear.sg.in]: @vigsterkr, I've made a force push, with all the corrections you've asked for. I hope you will now respond to my queries.
Running ```make && make meta_examples```  gave the following output https://pastebin.com/1cTQxeC4 

Which doesn't create any new files.

Comment (gf712) [reply examples/meta/src/kernel/linear.sg.in]: the files will be in `build/examples/meta/` and then you'll see the different language targets :)

Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg.in]: It should've been the case, but Visual studio code notified no creation of files. I even checked ```build/examples/meta/python/kernel/``` and I couldn't find ```linear.py``` as was expected

Comment (gf712) [reply examples/meta/src/kernel/linear.sg.in]: btw `make meta_examples` is enough to translate the examples, `make` just builds all targets. To compile the C++ target of your meta example it should be `make cpp-kernel-linear`

Comment (gf712) [reply examples/meta/src/kernel/linear.sg.in]: does visual studio have it's own build folder?

Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg.in]: I'm using docker for the ```make meta_examples``` commands and only using VSC to know the changes occured in the folder. I've ran all the ```mkdir build, cmake, make, etc``` through my docker image only

Comment (gf712) [reply examples/meta/src/kernel/linear.sg.in]: ok, can you see the file in your terminal?

Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg.in]: > btw `make meta_examples` is enough to translate the examples, `make` just builds all targets. To compile the C++ target of your meta example it should be `make cpp-kernel-linear`

Ran ```make cpp-kernel-linear```, got this
```make: *** No rule to make target 'cpp-kernel-linear'.  Stop.```

Comment (jonpsy) [reply examples/meta/src/kernel/linear.sg.in]: > ok, can you see the file in your terminal?

Nope, my docker /home and host /home are the same(share volume), so any changes in docker should be visible through VSC.

Comment (gf712) [reply examples/meta/src/kernel/linear.sg.in]: did you pass `-DENABLE_TESTING=ON  DISABLE_META_CPP=OFF` with `cmake`? For a complete list of these variables use `ccmake` please

Comment (gf712) [reply examples/meta/src/kernel/linear.sg.in]: from your pastebin your example wasn't built actually, so run `cmake ..` again

Comment (gf712) [reply examples/meta/src/kernel/linear.sg.in]: can you bring this conversation to irc please?

File: examples/meta/src/kernel/linear.sg.in [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+#![create_features]
+Features feats_train=features(traindat)
+Features feats_test=features(testdat)
+#![create_features]
+
+#![create_kernel]
+Kernel linear_kernel=kernel("LinearKernel")
+KernelNormalizer avg_diag_normalizer = AvgDiagKernelNormalizer()
+linear_kernel.set_normalizer(avg_diag_normalizer(1.2))
```
Comment (vigsterkr) [inline examples/meta/src/kernel/linear.sg.in]:
```suggestion
KernelNormalizer avg_diag_normalizer = AvgDiagKernelNormalizer(1.2)
linear_kernel.set_normalizer(avg_diag_normalizer)
```

**Pull request closed without merging**